web/themes/new_weather_theme/assets/js/components/combo-box.js

```diff
@@ -119,578 +120,687 @@ const comboTemplate = `
+
+      if (this.url) {
+        const result = {
+          text: this.input.value,
+          url: this.url,
+        };
+
+        this.saveSearchResult(result);
+        formEl.setAttribute("action", this.url);
+        return formEl.submit();
+      }
+
+      return this.getGeodataForKey(this.value).then((coordinates) => {
+        if (coordinates) {
+          const result = {
+            text: this.input.value,
+            url: `/point/${coordinates.lat}/${coordinates.lon}`,
+          };
+
+          this.saveSearchResult(result);
+
+          formEl.setAttribute("action", result.url);
+          formEl.submit();
         }
+      });
+    }
 
-        return Promise.reject(
-            new Error("No form ancestor element found for this combobox")
-        );
+    return Promise.reject(
+      new Error("No form ancestor element found for this combobox"),
+    );
+  }
+
+  /**
+   * Asynchronously fetches specific location data
+   * for a given search result, stashing it away in
+   * the cache (See ArcCache)
+   */
+  async cacheLocationGeodata(magicKey) {
+    if (!window.sessionStorage.getItem(magicKey)) {
+      const result = await getLocationGeodata(magicKey);
+      ArcCache.setItem(magicKey, result);
+    }
+  }
+
+  /**
+   * Attempts to retrieve location data for a search
+   * result by its magicKey from the cache.
+   * If not present, will make the Arc API call
+   * to fetch the data.
+   */
+  async getGeodataForKey(magicKey) {
+    const cached = ArcCache.getItem(magicKey);
+    if (!cached) {
+      return getLocationGeodata(magicKey);
     }
 
-    /**
-     * Asynchronously fetches specific location data
-     * for a given search result, stashing it away in
-     * the cache (See ArcCache)
-     */
-    async cacheLocationGeodata(magicKey){
-        if(!window.sessionStorage.getItem(magicKey)){
-            const result = await getLocationGeodata(magicKey);
-            ArcCache.setItem(magicKey, result);
-        }
+    return cached;
+  }
+
+  saveSearchResult(result) {
+    try {
+      // When loading previously-saved results, immediately remove any items for
+      // the same URL. We'll replace them with the new one. This way we don't
+      // end up having a whole list of the same place.
+      const saved = JSON.parse(
+        localStorage.getItem("wxgov_recent_locations") ?? "[]",
+      ).filter(({ url }) => url !== result.url);
+
+      // Put the new one at the front of the list.
+      saved.unshift(result);
+
+      localStorage.setItem(
+        "wxgov_recent_locations",
+        // We don't want to save everything forever. Just keep the most recent
+        // ten. We'll handle filtering down at display time.
+        JSON.stringify(saved.slice(0, 10)),
+      );
+    } catch (e) {
+      // Do nothing. If we're here, either the browser doesn't have local
+      // storage (which is unlikely in 2024, but possible), or it is
+      // disabled and we can't use it.
```

Comment:
In the future, would be good to log when this happens, but we don't have a current strategy for capturing this information.